base/abstractarray.jl

```diff
@@ -733,7 +733,7 @@ copyto!(dest::AbstractArray, src::AbstractArray) =
 
 function copyto!(::IndexStyle, dest::AbstractArray, ::IndexStyle, src::AbstractArray)
     destinds, srcinds = LinearIndices(dest), LinearIndices(src)
-    isempty(srcinds) || (first(srcinds) ∈ destinds && last(srcinds) ∈ destinds) ||
+    isempty(srcinds) || (checkbounds(Bool, destinds, first(srcinds)) && checkbounds(Bool, destinds, last(srcinds))) ||
```

Comment:
Maybe you could even simply pass `srcinds` to a single `checkbounds` call?

Otherwise LGTM but @timholy wrote that code and would probably know better.

Reply:
Tim is gone for two weeks without internet.

For now, I rather keep it a "mechanical" translation.